bench/wtperf/wtperf.c

```diff
@@ -755,19 +770,168 @@ worker(void *arg)
+			 * finished the actual insert.  Count failed search in
+			 * a random range as a "read".
+			 */
+			if (ret == WT_NOTFOUND)
+				break;
+			goto op_err;
+		case WORKER_MODIFY:
+			if ((ret = cursor->search(cursor)) == 0) {
+				if ((ret = cursor->get_value(
+				    cursor, &value)) != 0) {
+					lprintf(wtperf, ret, 0,
+					    "get_value in update.");
+					goto err;
+				}
+				/*
+				 * Copy as much of the previous value as is
+				 * safe, and be sure to NUL-terminate.
+				 */
+				strncpy(value_buf,
+				    value, opts->value_sz_max - 1);
+				if (workload->modify_delta != 0)
+					update_value_delta(thread,
+					    workload->modify_delta);
+				if (value_buf[0] == 'a')
+					value_buf[0] = 'b';
+				else
+					value_buf[0] = 'a';
+				if (opts->random_value)
+					randomize_value(thread,
+					    value_buf,
+					    workload->modify_delta);
+
+				/*
+				 * Distribute the modifications across whole
+				 * document.
+				 *
+				 * The distribution limits are same as MongoDB
+				 * that are imposed for an update operation to
+				 * be considered as modify or not?
+				 *
+				 * up to 10% of total modification size
+				 * 16 number of modifications per record
+				 *
+				 * NOTE: for records of size less than 160,
+				 * there is no distributed modifications.
+				 */
+
+				nmodifications = MAX_NUM_OF_MODIFICATIONS;
+
+				/* 10% of total size of the record */
+				total_modify_size = strlen(value) /
+				    MAX_SIZE_OF_MODIFICATION;
+
+				/*
+				 * Randomize the maximum modification size
+				 * and offset per modify. With randomization
+				 * of modification offset and size, we can
+				 * produce more number of modifications even
+				 * if the same record is modified again.
+				 */
+				rand_val = __wt_random(&thread->rnd);
+				modify_size =
+				    total_modify_size / (size_t)nmodifications;
+				if (modify_size != 0)
+					modify_size =
+					    (size_t)rand_val % modify_size;
+
+				/*
+				 * Offset location difference between
+				 * modifications
+				 */
+				modify_offset =
+				    strlen(value) / (size_t)nmodifications;
+				if (modify_offset != 0)
+					modify_offset =
+					    (size_t)rand_val % modify_offset;
+
+				/*
+				 * Make sure the offset is more than size,
+				 * otherwise modifications don't spread
+				 * properly.
+				 */
+				if (modify_offset < modify_size)
+				    modify_offset = modify_size + 1;
+
+				for (iter = (size_t)nmodifications;
+					 iter > 0;
+					 iter--)
+					strncpy(
+					    &value_buf[(iter * modify_offset) -
+					    modify_offset],
+					    &value_buf[(iter * modify_offset) -
+					    modify_size],
+					    modify_size);
+
+				/*
+				 * Increase the number of modifications, so
+				 * that normal modify operations gets
+				 * succeeded.
+				 */
+				if (!workload->modify_force_update)
+					nmodifications++;
+
+				oldv.data = value;
+				oldv.size = strlen(value);
+
+				newv.data = value_buf;
+				newv.size = strlen(value_buf);
```

Comment:
These could be combined to `oldv.data = newv.data = value_buf;`, and similar for size.

Reply:
Actually **oldv** is pointing to **value** and the **newv** is pointing to **value_buf**. In the current test, both the sizes are the same, but in case if there is any difference because of modify_delta, we cannot assign the same in the single line.